### PR TITLE
Add more ConcurrentSnapshotsIT tests

### PR DESCRIPTION
**Pull request by matriv**
Relates: #13613

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/snapshots/ConcurrentSnapshotsIT.java [hunk 1]
```diff
@@ -805,6 +856,14 @@ private CompletableFuture<SnapshotInfo> startAndBlockFailingFullSnapshot(String
         return future;
     }
 
+    private CompletableFuture<AcknowledgedResponse> startAndBlockOnDeleteSnapshot(String repoName, String snapshotName) throws Exception {
+        final String masterName = cluster().getMasterName();
+        mockRepo(repoName, masterName).setBlockOnDeleteIndexFile();
```
Comment (matriv) [inline server/src/test/java/org/elasticsearch/snapshots/ConcurrentSnapshotsIT.java:861]: if we have `setBlockOnAnyFiles()` then `getRepositoryData()` is already blocked and you don't have in progress deletions.

Comment (mfussenegger) [reply server/src/test/java/org/elasticsearch/snapshots/ConcurrentSnapshotsIT.java:861]: Just fyi: I think in ES this works with the data block due to their repository data cache.

**Review by mfussenegger — APPROVED** [review]

File: server/src/test/java/org/elasticsearch/snapshots/ConcurrentSnapshotsIT.java (hunk 1 above)
Comment (matriv) [reply server/src/test/java/org/elasticsearch/snapshots/ConcurrentSnapshotsIT.java:861]: yep, I was just trying to figure out a workaround, it affects other tests, but will try to figure out something, maybe introduce some other block type in the MockRepository.

**Comment by matriv** [comment]
Unrelated: `org.elasticsearch.indices.recovery.IndexRecoveryIT.testRerouteRecovery` failure

**Pull request merged**